api/src/services/db-doc.js

```diff
@@ -82,29 +100,25 @@ module.exports = {
 
   // db-doc GET requests with `open_revs` return a list of requested revisions of the requested doc id
   filterOfflineOpenRevsRequest: (userCtx, params, query) => {
-    return Promise
-      .all([
-        getStoredDoc(params, 'GET', query),
-        authorization.getAuthorizationContext(userCtx)
-      ])
-      .then(([ storedDocs, authorizationContext ]) => {
-        return storedDocs.filter(storedDoc => {
-          if (!storedDoc.ok) {
-            return false;
-          }
-
-          const viewResults = authorization.getViewResults(storedDoc.ok);
-          return authorization.allowedDoc(storedDoc.ok._id, authorizationContext, viewResults) ||
-                 authorization.isDeleteStub(storedDoc.ok);
-        });
+    let stored;
```

Comment:
Nit: I strongly prefer `const` over `let` and in this case it's only written to once so `const` is doable. In this case I think all you need to do is move the second `.then` into the first `.then` and the scope will work.

Reply:
Updated. I probably hate nesting promises more than `let`, but this didn't turn out so bad :)